packages/gestalt/src/Masonry/defaultLayout.test.ts

```diff
@@ -247,3 +247,69 @@ describe.each([undefined, getColumnSpanConfig])('default layout tests', (_getCol
     ]);
   });
 });
+
+describe('loadingStateItems', () => {
+  test("uses the loadingStateItem's height", () => {
+    const measurementStore = new MeasurementStore<Record<any, any>, number>();
+    const positionCache = new MeasurementStore<Record<any, any>, Position>();
+    const loadingStateItems: ReadonlyArray<Item> = [
+      { 'name': 'Pin 0', 'height': 100 },
+      { 'name': 'Pin 1', 'height': 120 },
+      { 'name': 'Pin 2', 'height': 80 },
+      { 'name': 'Pin 3', 'height': 100 },
+    ];
+
+    const layout = defaultLayout({
+      align: 'start',
+      measurementCache: measurementStore,
+      positionCache,
+      layout: 'basic',
+      minCols: 2,
+      rawItemCount: loadingStateItems.length,
+      width: 8000,
+      renderLoadingState: true,
+    });
+
+    expect(layout(loadingStateItems)).toEqual([
+      { top: 0, height: 100, left: 0, width: 236 },
+      { top: 0, height: 120, left: 250, width: 236 },
+      { top: 0, height: 80, left: 500, width: 236 },
+      { top: 0, height: 100, left: 750, width: 236 },
+    ]);
+  });
+
+  test('uses the measurementCache height when not a loadingStateitem', () => {
+    const measurementStore = new MeasurementStore<Record<any, any>, number>();
+    const positionCache = new MeasurementStore<Record<any, any>, Position>();
+    const items: ReadonlyArray<Item> = [
+      { 'name': 'Pin 0', 'height': 100 },
+      { 'name': 'Pin 1', 'height': 120 },
+      { 'name': 'Pin 2', 'height': 80 },
+      { 'name': 'Pin 3', 'height': 100 },
+    ];
+    items.forEach((item: any) => {
```

Comment:
added this test because i noticed that my `isLoadingStateItem` check would always return true if an item had a height key (even if it should not render a loading state item)

updated `isLoadingStateItem` to also check that `renderLoadingState` is true